src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -2654,6 +2654,30 @@ public void test_merge_with_message() throws Exception {
 
         assertEquals("Custom message merge failed. Should have set custom merge message.", mergeMessage, resultMessage);
     }
+    
+    public void test_changelog_with_merge_commit() throws Exception {
+        w.init();
+        w.commitEmpty("init");
+
+        // First commit to branch1
+        w.git.branch("branch1");
+        w.git.checkout("branch1");
+        w.touch("file1", "content1");
+        w.git.add("file1");
+        w.git.commit("commit1");
+        String commitSha1 = w.git.revParse("HEAD").name();
+
+        // Merge branch1 into master
+        w.git.checkout("master");
+        String mergeMessage = "Merge message to be tested.";
+        w.git.merge().setMessage(mergeMessage).setGitPluginFastForwardMode(MergeCommand.GitPluginFastForwardMode.NO_FF).setRevisionToMerge(w.git.getHeadRev(w.repoPath(), "branch1")).execute();
+        // Obtain last commit message
+        String mergeSha1 = w.git.revParse("HEAD").name();
+        check_changelog_sha1(w.git.changelog().listMerges(true),mergeSha1, "master");
+        check_changelog_sha1(w.git.changelog().listMerges(false),commitSha1, "master");
+        check_changelog_sha1(w.git.changelog(),commitSha1, "master");
```

Comment:
@MarkEWaite  The above 2 tests fail with JGit and pass with CliGit.
The failure for this simple scenario is that JGit revision walker sees the HEAD commit (which is a merge commit). the ignore it becuase it has more then 1 parent (so far, as expected). But then when it should look for the previous commit in the history, it fails, as if there are no commits prior to the HEAD.
Whereas CliGit return the commit of "commit1" introduced in branch1 before the merge.
I think this is a JGit bug, and I don't quite understand the walker logic to provide further hints.
Was hoping you could shed some light on this.